source/mqtt5_client.c

```diff
@@ -2018,34 +2017,56 @@ PyObject *aws_py_mqtt5_client_get_stats(PyObject *self, PyObject *args) {
         return NULL;
     }
 
-    /* These must be DECREF'd when function ends */
+    /* These must be DECREF'd when function ends on error */
     PyObject *result = NULL;
+    PyObject *incomplete_operation_count_obj = NULL;
+    PyObject *incomplete_operation_size_obj = NULL;
+    PyObject *unacked_operation_count_obj = NULL;
+    PyObject *unacked_operation_size_obj = NULL;
 
     struct aws_mqtt5_client_operation_statistics stats;
     AWS_ZERO_STRUCT(stats);
 
     aws_mqtt5_client_get_stats(client->native, &stats);
 
-    result = PyObject_CallFunction(
-        get_stats_callback_fn_py,
-        "(KKKK)",
-        /* K */ (unsigned long long)stats.incomplete_operation_count,
-        /* K */ (unsigned long long)stats.incomplete_operation_size,
-        /* K */ (unsigned long long)stats.unacked_operation_count,
-        /* K */ (unsigned long long)stats.unacked_operation_size);
+    result = PyTuple_New(4);
     if (!result) {
-        PyErr_WriteUnraisable(PyErr_Occurred());
         goto done;
     }
 
+    incomplete_operation_count_obj = PyLong_FromUnsignedLongLong((unsigned long long)stats.incomplete_operation_count);
+    if (!incomplete_operation_count_obj) {
+        goto done;
+    }
```

Comment:
See my next comment explaining WHY this should probably look like:

```
num_obj = PyLong_FromUnsignedLongLong(stats.incomplete_operation_count);
if (!num_obj) {
    goto done;
}
PyTuple_SET_ITEM(result, 0, num_obj); /* steals reference to num_obj */

num_obj = PyLong_FromUnsignedLongLong(stats.incomplete_operation_size);
if (!num_obj) {
    goto done;
}
PyTuple_SET_ITEM(result, 1, num_obj); /* steals reference to num_obj */
...
```

or maybe even simpler:

```
PyTuple_SET_ITEM(result, 0, PyLong_FromUnsignedLongLong(stats.incomplete_operation_count));
if (PyTuple_GET_ITEM(result, 0) == NULL) {
    goto done;
}

PyTuple_SET_ITEM(result, 1, PyLong_FromUnsignedLongLong(stats.incomplete_operation_size));
if (PyTuple_GET_ITEM(result, 1) == NULL) {
    goto done;
}
...
```